File: study/2023/23.02.07/EJH/최소직사각형.java
import java.util.Arrays;

public class 최소직사각형 {
    public int solution(int[][] sizes) {
        int answer = 0;
        int max_v=0;
        int max_h=0;
        for(int i=0;i<sizes.length;i++){
            Arrays.sort(sizes[i]);
            max_v=Math.max(max_v,sizes[i][0]);
            max_h=Math.max(max_h,sizes[i][1]);
        }
        return answer=max_v*max_h;
    }
}

File: study/2023/23.02.07/HJH/최소직사각형.js
function solution(sizes) {
    const rotated = sizes.map(([w, h]) => w < h ? [h, w] : [w, h]);
    let maxSize = [0, 0];
    rotated.forEach(([w, h])=> {
        if (w > maxSize[0]) maxSize[0] = w;
        if (h > maxSize[1]) maxSize[1] = h;
        // maxSize[0] = Math.max(maxSize[0], w);
        // maxSize[1] = Math.max(maxSize[1], h);
    })
    return maxSize[0]*maxSize[1];
}
